internal/commands/vaultsecrets/secrets/create.go

```diff
@@ -0,0 +1,168 @@
+				Command: heredoc.New(ctx.IO, heredoc.WithNoWrap()).Must(`
+				$ hcp vault-secrets secrets create secret_3 --app-name test-app --secret_file=/tmp/secrets2.txt
+				`),
+			},
+		},
+		Args: cmd.PositionalArguments{
+			Args: []cmd.PositionalArgument{
+				{
+					Name:          "SECRET_NAME",
+					Documentation: "The name of the secret to create.",
+				},
+			},
+		},
+		Flags: cmd.Flags{
+			Local: []*cmd.Flag{
+				{
+					Name:         "data-file",
+					DisplayValue: "DATA_FILE_PATH",
+					Description:  "Absolute path to the secrets data file.",
+					Value:        flagvalue.Simple("", &opts.SecretFilePath),
+				},
+			},
+		},
+		RunF: func(c *cmd.Command, args []string) error {
+			opts.AppName = appName
+			opts.SecretName = args[0]
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return createRun(opts)
+		},
+	}
+
+	return cmd
+}
+
+type CreateOpts struct {
+	Ctx     context.Context
+	Profile *profile.Profile
+	IO      iostreams.IOStreams
+	Output  *format.Outputter
+
+	AppName              string
+	SecretName           string
+	SecretValuePlaintext string
+	SecretFilePath       string
+	PreviewClient        preview_secret_service.ClientService
+	Client               secret_service.ClientService
+}
+
+func createRun(opts *CreateOpts) error {
+	if err := readPlainTextSecret(opts); err != nil {
+		return err
+	}
+
+	req := secret_service.NewCreateAppKVSecretParamsWithContext(opts.Ctx)
+	req.LocationOrganizationID = opts.Profile.OrganizationID
+	req.LocationProjectID = opts.Profile.ProjectID
+	req.AppName = opts.AppName
+
+	req.Body = secret_service.CreateAppKVSecretBody{
+		Name:  opts.SecretName,
+		Value: opts.SecretValuePlaintext,
+	}
+
+	resp, err := opts.Client.CreateAppKVSecret(req, nil)
+	if err != nil {
+		return fmt.Errorf("failed to create secret with name: %s - %w", opts.SecretName, err)
+	}
+	return opts.Output.Display(newDisplayer(true, resp.Payload.Secret))
+}
+
+func readPlainTextSecret(opts *CreateOpts) error {
+	if opts.SecretValuePlaintext != "" {
+		return nil
+	}
+
+	if opts.SecretFilePath == "" {
+		return errors.New("data file path is required")
+	}
+
+	if opts.SecretFilePath == "-" {
+		plaintextSecretBytes, err := io.ReadAll(opts.IO.In())
+		if err != nil {
+			return fmt.Errorf("failed to read the plaintext secret: %w", err)
+		}
+
+		if len(plaintextSecretBytes) == 0 {
+			return errors.New("secret value cannot be empty")
+		}
+		opts.SecretValuePlaintext = string(plaintextSecretBytes)
+		return nil
+	}
+
+	fileInfo, err := os.Stat(opts.SecretFilePath)
+	if err != nil {
+		return fmt.Errorf("%s failed to get data file info: %w", opts.IO.ColorScheme().FailureIcon(), err)
+	}
+
+	if fileInfo.Size() == 0 {
+		return fmt.Errorf("%s data file cannot be empty", opts.IO.ColorScheme().FailureIcon())
+	}
+
+	data, err := os.ReadFile(opts.SecretFilePath)
+	if err != nil {
+		return fmt.Errorf("%s unable to read the data file: %w", opts.IO.ColorScheme().FailureIcon(), err)
```

Comment:
nit: I would not include the failure icon in this error. If you want the icon, I would put it where readPlainTextSecret is called so it applies to all

Reply:
actually, the suggestion is great. Will love it up.